src/pages/workspaces/workspace/analysis/modals/AzureComputeModal.test.js

```diff
@@ -0,0 +1,164 @@
+}))
+const onSuccess = jest.fn()
+const defaultModalProps = {
+  onSuccess, onDismiss: jest.fn(), onError: jest.fn(),
+  currentRuntime: undefined, currentDisk: undefined, tool: toolLabels.JupyterLab, workspace: defaultAzureWorkspace,
+  location: testAzureDefaultRegion
+}
+
+
+const defaultAjaxImpl = {
+  Runtimes: {
+    runtime: () => ({
+      details: jest.fn()
+    })
+  },
+  Buckets: { getObjectPreview: () => Promise.resolve({ json: () => Promise.resolve(imageDocs) }) },
+  Disks: {
+    disk: () => ({
+      details: jest.fn()
+    })
+  },
+  Metrics: {
+    captureEvent: () => jest.fn()
+  }
+}
+
+const verifyEnabled = item => expect(item).not.toHaveAttribute('disabled')
+
+describe('AzureComputeModal', () => {
+  beforeAll(() => {
+  })
+
+  beforeEach(() => {
+    // Arrange
+    Ajax.mockImplementation(() => ({
+      ...defaultAjaxImpl
+    }))
+  })
+
+  afterEach(() => {
+    jest.clearAllMocks()
+  })
+
+  const getCreateButton = () => screen.getByText('Create')
+  it('renders correctly with minimal state', async () => {
+    // Arrange
+
+    // Act
+    await act(async () => await render(h(AzureComputeModalBase, defaultModalProps)))
+
+    // Assert
+    verifyEnabled(getCreateButton())
+    screen.getByText('Azure Cloud Environment')
+  })
+
+  it('sends the proper leo API call in default create case (no runtimes or disks)', async () => {
+    // Arrange
+    const createFunc = jest.fn()
+    const runtimeFunc = jest.fn(() => ({
+      create: createFunc,
+      details: jest.fn()
+    }))
+    Ajax.mockImplementation(() => ({
+      ...defaultAjaxImpl,
+      Runtimes: {
+        runtimeV2: runtimeFunc
+      }
+    }))
+
+    // Act
+    await act(async () => {
+      await render(h(AzureComputeModalBase, defaultModalProps))
+      await userEvent.click(getCreateButton())
+    })
+
+    // Assert
+    const labels = {
+      saturnWorkspaceNamespace: defaultModalProps.workspace.workspace.namespace,
+      saturnWorkspaceName: defaultModalProps.workspace.workspace.name
+    }
+    expect(runtimeFunc).toHaveBeenCalledWith(defaultModalProps.workspace.workspace.workspaceId, expect.anything())
+    expect(createFunc).toHaveBeenCalledWith(expect.objectContaining({
+      labels,
+      disk: expect.objectContaining({
+        labels,
+        name: expect.anything()
+      }),
+      machineSize: defaultAzureMachineType
+    }))
+
+    expect(onSuccess).toHaveBeenCalled()
+  })
+
+  it('renders default cost estimate', async () => {
+    // Arrange
+    const expectedComputeCost = 0.15
+    const expectedDiskCost = 0.20
+
+    asMockedFn(getAzureComputeCostEstimate).mockReturnValue(expectedComputeCost)
+    asMockedFn(getAzureDiskCostEstimate).mockReturnValue(expectedDiskCost)
+
+    // Act
+    await act(async () => {
+      await render(h(AzureComputeModalBase, defaultModalProps))
+    })
+
+    // Assert
+    expect(screen.getAllByText(formatUSD(expectedComputeCost))) // Currently stopped and running are the same cost.
```

Comment:
an assertion with getAllByText typically asserts the length of the results array to be 0, 1, or the expected count.  As is these will always pass since getAllByText returns an array, which even when empty is "truthy".